packages/wallet-sdk/src/derive.ts

```diff
@@ -197,9 +201,38 @@ const selectUsernameForAccount = async ({
     }
   }
   // use wallet derivation for accounts without username
+  // or without network parameter (offline)
   return { username: undefined, derivationType: DerivationType.Wallet };
 };
 
+export const fetchUsernameForAccountByDerivationType = async ({
+  rootNode,
+  index,
+  derivationType,
+  network,
+}: {
+  rootNode: BIP32Interface;
+  index: number;
+  derivationType: DerivationType.Wallet | DerivationType.Data;
+  network?: StacksNetwork;
+}): Promise<{
+  username: string | undefined;
+}> => {
+  // try to find existing usernames owned by given derivation path
+  if (network) {
+    const txVersion = whenChainId(network.chainId)({
+      [ChainID.Mainnet]: TransactionVersion.Mainnet,
+      [ChainID.Testnet]: TransactionVersion.Testnet,
+    });
+    const privateKey = derivePrivateKeyByType({ rootNode, index, derivationType });
+    const address = getAddressFromPrivateKey(privateKey, txVersion);
+    const username = await fetchFirstName(address, network);
+    return { username };
+  } else {
+    return { username: undefined };
+  }
+};
+
```

Comment:
Should we make `network` a require parameter, as it's pretty pointless to use this function without one.

Reply:
Fixed in 9b261525dce70f4e4463eccc2b1feeed178f1458